Source/Core/DiscIO/FileSystemGCWii.cpp

```diff
@@ -252,36 +269,43 @@ void CFileSystemGCWii::InitFileSystem()
 {
 	m_Initialized = true;
 	u32 const shift = GetOffsetShift();
+	u32 read_buffer;
 
 	// read the whole FST
-	u64 FSTOffset = static_cast<u64>(m_rVolume->Read32(0x424, m_Wii)) << shift;
-	// u32 FSTSize     = Read32(0x428);
-	// u32 FSTMaxSize  = Read32(0x42C);
-
+	if (!m_rVolume->Read32(0x424, &read_buffer, m_Wii))
+		return;
+	u64 FSTOffset = static_cast<u64>(read_buffer) << shift;
 
 	// read all fileinfos
-	SFileInfo Root
-	{
-		m_rVolume->Read32(FSTOffset + 0x0, m_Wii),
-		static_cast<u64>(FSTOffset + 0x4) << shift,
-		m_rVolume->Read32(FSTOffset + 0x8, m_Wii)
-	};
+	SFileInfo root;
+	if (!m_rVolume->Read32(FSTOffset + 0x0, &read_buffer, m_Wii))
```

Comment:
I think it's clearer to read all those 3 values at once and to generate the SFileInfo struct afterwards.
```
if (!read() || !read() || !read()) return;
...
```